modules/waardexBidAdapter.js

```diff
@@ -1,217 +1,294 @@
-    return false;
+};
+
+const getBidRequestsToSend = validBidRequests => {
+  return validBidRequests.map(getBidRequestToSend);
+};
+
+const getBidRequestToSend = validBidRequest => {
+  const result = {
+    bidId: validBidRequest.bidId,
+    bidfloor: parseFloat(validBidRequest.params.bidfloor) || 0,
+    position: parseInt(validBidRequest.params.position) || 1,
+    instl: parseInt(validBidRequest.params.instl) || 0,
+  };
+
+  if (validBidRequest.mediaTypes[BANNER]) {
+    result[BANNER] = createBannerObject(validBidRequest.mediaTypes[BANNER]);
   }
 
-  return Boolean(bid.width && bid.height && bid.ad);
-}
-
-/**
- * @param {Object} serverBid
- *
- * @returns {Object|null}
- * */
-function createBid(serverBid) {
-  const bid = {
-    requestId: serverBid.id,
-    cpm: serverBid.price,
+  if (validBidRequest.mediaTypes[VIDEO]) {
+    result[VIDEO] = createVideoObject(validBidRequest.mediaTypes[VIDEO], validBidRequest.params);
+  }
+
+  return result;
+};
+
+const createBannerObject = banner => {
+  return {
+    sizes: transformSizes(banner.sizes),
+  };
+};
+
+const transformSizes = requestSizes => {
+  let result = [];
+
+  if (Array.isArray(requestSizes) && !Array.isArray(requestSizes[0])) {
+    result[0] = {
+      width: parseInt(requestSizes[0], 10) || 0,
+      height: parseInt(requestSizes[1], 10) || 0,
+    };
+  } else if (Array.isArray(requestSizes) && Array.isArray(requestSizes[0])) {
+    result = requestSizes.map(item => {
+      return {
+        width: parseInt(item[0], 10) || 0,
+        height: parseInt(item[1], 10) || 0,
+      }
+    });
+  }
+
+  return result;
+};
+
+const createVideoObject = (videoMediaTypes, videoParams) => {
+  return {
+    w: utils.deepAccess(videoMediaTypes, 'playerSize')[0][0],
+    h: utils.deepAccess(videoMediaTypes, 'playerSize')[0][1],
+    mimes: utils.getBidIdParameter('mimes', videoParams) || ['application/javascript', 'video/mp4', 'video/webm'],
+    minduration: utils.getBidIdParameter('minduration', videoParams) || 0,
+    maxduration: utils.getBidIdParameter('maxduration', videoParams) || 500,
+    protocols: utils.getBidIdParameter('protocols', videoParams) || [2, 3, 5, 6],
+    startdelay: utils.getBidIdParameter('startdelay', videoParams) || 0,
+    placement: utils.getBidIdParameter('placement', videoParams) || videoMediaTypes.context === 'outstream' ? 3 : 1,
+    skip: utils.getBidIdParameter('skip', videoParams) || 1,
+    skipafter: utils.getBidIdParameter('skipafter', videoParams) || 0,
+    minbitrate: utils.getBidIdParameter('minbitrate', videoParams) || 0,
+    maxbitrate: utils.getBidIdParameter('maxbitrate', videoParams) || 3500,
+    delivery: utils.getBidIdParameter('delivery', videoParams) || [2],
+    playbackmethod: utils.getBidIdParameter('playbackmethod', videoParams) || [1, 2, 3, 4],
+    api: utils.getBidIdParameter('api', videoParams) || [2],
+    linearity: utils.getBidIdParameter('linearity', videoParams) || 1
```

Comment:
@yegorWaardex - I know it's too late for this PR, but if I'm reading it correctly, this code breaks the Prebid rule that mediaType video parameters must take precedence over bidder-specific video params.

Our goal is that publishers should only have to specify video parameters in one place -- at the mediatype level. They should not have to copy video params into each bidder's param section.

If I'm misunderstanding the code here, please forgive. If I'm right, please submit another PR to check the video mediatype params first.

thanks